src/mapblock.h

```diff
@@ -73,15 +73,12 @@ class MapBlock
 		m_orphan = true;
 	}
 
-	void reallocate()
-	{
-		for (u32 i = 0; i < nodecount; i++)
-			data[i] = MapNode(CONTENT_IGNORE);
-		raiseModified(MOD_STATE_WRITE_NEEDED, MOD_REASON_REALLOCATE);
-	}
-
 	MapNode* getData()
 	{
+		if (m_is_mono_block) {
+			reallocate(nodecount, data[0]);
```

Comment:
Not much else I can do here, unless I remove this method.

Reply:
How many callers does this have?

Reply:
As far as I can see only MeshUpdateQueue::fillDataFromMapBlocks, i.e. on the client, where to do not have monoblocks anyway.